app/models/manageiq/providers/kubernetes/container_manager/scanning/job.rb

```diff
@@ -215,6 +211,16 @@ def finish(*_args)
 
   alias_method :abort_job, :cleanup
 
+  def cancel(*args)
+    _log.info "Job Canceling"
+    if self.state != "canceling" # ensure change of states
+      signal :cancel
+    else
+      unqueue_all_signals
+      queue_signal(:abort_job, "Job Canceled", "cancel", *args)
```

Comment:
@moolitayer I am not sure that we want to do that. if we call ```process_cancel``` it will not ```set_status``` and the job will look to the user like it has finished successfully (The only clue will be in the log were it will be written that the job was canceled)